plugins/goodix-fp/fu-goodixfp-common.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016-2018 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2020 boger wang <boger@goodix.com>
+ * 
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <glib-object.h>
+/*
+ *  protocol
+ */
+#define GX_CMD_ACK          0xAA
+#define GX_CMD_VERSION      0xD0
+#define GX_CMD_RESET        0xB4
+#define GX_CMD_UPGRADE      0x80
+#define GX_CMD_UPGRADE_INIT 0x00
+#define GX_CMD_UPGRADE_DATA 0x01
+#define GX_CMD1_DEFAULT     0x00
+
+#define PACKAGE_HEADER_SIZE (8)
+#define PACKAGE_CRC_SIZE (4)
+
+typedef struct _gxfp_version_info
+{
+  guint8 format[2];
+  guint8 fwtype[8];
+  guint8 fwversion[8];
+  guint8 customer[8];
+  guint8 mcu[8];
+  guint8 sensor[8];
+  guint8 algversion[8];
+  guint8 interface[8];
+  guint8 protocol[8];
+  guint8 flashVersion[8];
+  guint8 reserved[62];
+} gxfp_version_info_t, *pgxfp_version_info_t;
```

Comment:
in fwupd we tend to just do:

    typedef struct {
        foo
    } GxfpVersionInfo;